internal/sink/sink.go

```diff
@@ -213,3 +244,236 @@ func (v1 *WebhookV1) onAttempt(request *http.Request, event string) retry.OnAtte
 
 	}
 }
+
+func (k Kafkas) Update(l ancla.Register) error {
+	return nil
+}
+
+// TODO: probably get rid of urls
+func (k Kafkas) Send(urls *ring.Ring, secret string, acceptType string, msg *wrp.Message) error {
+	//TODO: is this how we want to set this up?
+	//or do we want to only send to specific kafkas in the list based on an id
+	for _, kafka := range k {
+		err := kafka.send(secret, acceptType, msg)
+		return err
+	}
+	return nil
+}
+
+func (k *Kafka) send(secret string, acceptType string, msg *wrp.Message) error {
+
+	defer func() {
+		if r := recover(); nil != r {
+			// s.droppedPanic.Add(1.0)
+			k.logger.Error("goroutine send() panicked", zap.String("id", k.id), zap.Any("panic", r))
+		}
+		// s.workers.Release()
+		// s.currentWorkersGauge.Add(-1.0)
+	}()
+
+	payload := msg.Payload
+	body := payload
+
+	// Use the internal content type unless the accept type is wrp
+	contentType := msg.ContentType
+	switch acceptType {
+	case "wrp", wrp.MimeTypeMsgpack, wrp.MimeTypeWrp:
+		// WTS - We should pass the original, raw WRP event instead of
+		// re-encoding it.
+		contentType = wrp.MimeTypeMsgpack
+		//TODO: do we want to use the wrp encoder or the sarama encoder?
+		buffer := bytes.NewBuffer([]byte{})
+		encoder := wrp.NewEncoder(buffer, wrp.Msgpack)
+		encoder.Encode(msg)
+		body = buffer.Bytes()
+	}
+
+	// Create a new Kafka producer
+	producer, err := sarama.NewSyncProducer(k.brokerAddr, k.config)
```

Comment:
I think we want to cache the producer and not recreate it for every incoming event